bumble/transport/__init__.py

```diff
@@ -40,9 +40,16 @@ def _wrap_transport(transport: Transport) -> Transport:
     # If BUMBLE_SNOOPER is set, try to automatically create a snooper.
     if snooper_spec := os.getenv('BUMBLE_SNOOPER'):
         try:
-            return SnoopingTransport.create_with(
+            wrapped_transport = SnoopingTransport.create_with(
                 transport, create_snooper(snooper_spec)
             )
+
+            # Transfer and extend metadata
+            if hasattr(transport, "source"):
```

Comment:
It may be simpler to just add a `metadata`  `@property` on the `SnoopingTransport.Source` class that returns the wrapped source's metadata (no copy needed, it's Ok to pass the wrapped object's properties directly up).

Reply:
Agree, that's better. Just done it.